m4/libevent_openssl.m4

```diff
@@ -3,6 +3,20 @@ dnl OpenSSL support
 AC_DEFUN([LIBEVENT_OPENSSL], [
 AC_REQUIRE([NTP_PKG_CONFIG])dnl
 
+case "$host_os" in
+    darwin*)
+    dnl when compiling for darwin, attempt to find OpenSSL using brew.
+    AC_CHECK_PROG([BREW],brew, brew)
+    if test x$BREW = xbrew; then
+        openssl_prefix=$($BREW --prefix openssl 2>/dev/null)
+        if test xopenssl_prefix != x; then
+            PKG_CONFIG_PATH="$openssl_prefix/lib/pkgconfig:$PKG_CONFIG_PATH"
+            export PKG_CONFIG_PATH
```

Comment:
export does not looks good, how about calling pkg-config explicitly with --with-path=path?

Reply:
Could you clarify what you mean. From what I can see, `pkg-config` does not have a `--with-path=` option. Or do you mean passing a path into `./configure`?

Reply:
Interesting, which version of pkg-config do you have?

```
$ pkg-config --version
1.7.3
$ pkg-config -h |& fgrep with-path
  --with-path=path                  adds a directory to the search path
```

I'm pretty sure that any version of pkg-config has an analog for PKG_CONFIG_PATH env variable

Reply:
```bash
➜ pkg-config --version          
0.29.2
➜ pkg-config -h | grep with-path
➜
```

Interesting. Given the `1.7.3` version number, I assume your `pkg-config` is actually https://github.com/pkgconf/pkgconf and not https://freedesktop.org/wiki/Software/pkg-config/?

Reply:
Indeed, interesting.

Mine is:

```
$ pacman -Qi pkg-config
URL             : https://git.sr.ht/~kaniini/pkgconf
```

Which is:

>pkgconf is a program which helps to configure compiler and linker flags for development libraries. **It is similar to pkg-config from freedesktop.org.**

So yes, not freedesktop version, and [later](https://man.openbsd.org/pkg-config.1) does not have option for `PKG_CONFIG_PATH`

Ok, let's keep env var, but with this information in comment

Reply:
> Ok, let's keep env var, but with this information in comment

@azat just to clairfy, you want some notes about pkg-config alongside the exporting? 
I can add something along those lines and then take this PR out of draft.

Reply:
Yes, I want to add a comment why the `export` is used (so that nobody won't try change this later)

Reply:
>I can add something along those lines and then take this PR out of draft.

@fanquake no luck with cmake? ok, no problem, I can finish this by myself

Reply:
Ok, cmake does not uses pkg-config, let's ignore for now